src/EPR.Calculator.API/Services/TransposePomAndOrgDataService.cs

```diff
@@ -130,5 +177,56 @@ public void Transpose(CalcResultsRequestDto resultsRequestDto)
             }
         }
+
+        private IEnumerable<OrganisationDetails> GetOrganisationDetailsBySubmissionPeriod()
+        { 
+            OrganisationsBySubmissionPeriod = (from org in OrganisationsList
+                                               join sub in SubmissionPeriodDetails on
+                                               org.SubmissionPeriodDescription equals sub.SubmissionPeriodDesc
+                                               select new OrganisationDetails
+                                               {
+                                                   OrganisationId = org.OrganisationId,
+                                                   OrganisationName = org.OrganisationName,
+                                                   SubmissionPeriodDescription = org.SubmissionPeriodDescription,
+                                                   SubmissionPeriod = sub.SubmissionPeriod,
+                                                   SubsidaryId = org.SubsidaryId
+                                               }).ToList();
+
+            return OrganisationsBySubmissionPeriod;
+           
+        }
+
+        private void GetAllOrganisationsBasedonRunId(int runId)
+        {
+            OrganisationsList = [.. (from run in context.CalculatorRuns join
+                 org in context.CalculatorRunOrganisationDataDetails on run.CalculatorRunOrganisationDataMasterId equals org.CalculatorRunOrganisationDataMasterId
+                    where (run.Id == runId && org.OrganisationName != null)
+                    select new OrganisationDetails
+                    {
+                        OrganisationId = org.OrganisationId,
+                        OrganisationName = org.OrganisationName,
+                        SubmissionPeriodDescription = org.SubmissionPeriodDesc,
+                        SubsidaryId = org.SubsidaryId
+                    }).Distinct()];
+        }
+       
+
+        private string? GetLatestOrganisationName(int orgId)
+        {
+            if (OrganisationsBySubmissionPeriod is null) return string.Empty;
+
+            var organisations = OrganisationsBySubmissionPeriod.Where(t => t.OrganisationId == orgId && t.SubsidaryId == null).OrderByDescending(t => t.SubmissionPeriod?.Replace(PeriodSeparator, string.Empty)).ToList();
+
+            var orgName = organisations?.FirstOrDefault(t => t.OrganisationId == orgId)?.OrganisationName;
+            return string.IsNullOrWhiteSpace(orgName) ? OrganisationsList.FirstOrDefault(t => t.OrganisationId == orgId)?.OrganisationName : orgName;
+        }
+
+        private string? GetLatestSubsidaryName(int orgId, string? subsidaryId)
+        {
+            if (OrganisationsBySubmissionPeriod is null) return string.Empty;
+            var subsidaries = OrganisationsBySubmissionPeriod.Where(t => t.OrganisationId == orgId && t.SubsidaryId == subsidaryId).OrderByDescending(t => t.SubmissionPeriod?.Replace(PeriodSeparator, string.Empty)).ToList();
```

Comment:
Can you pass OrganisationsBySubmissionPeriod as a parameter?

Reply:
updated as per comments